src/nsrt_learning.py

```diff
@@ -34,52 +33,41 @@ def learn_nsrts_from_data(dataset: Dataset, predicates: Set[Predicate],
     segments = [seg for segs in segmented_trajs for seg in segs]
 
     # STEP 3: Cluster the data by effects, jointly producing one STRIPSOperator,
-    #         OptionSpec, and Partition (data store) per cluster. These items
-    #         then used to initialize _NSRTIntermediateData objects (NIDs).
+    #         NSRTDatastore, and OptionSpec per cluster. These items are then
+    #         used to initialize PartialNSRTAndDatastore objects (PNADs).
     #         Note: The OptionSpecs here are extracted directly from the data.
     #         If we are doing option learning, then the data will not contain
     #         options, and so the option_spec fields are just the specs of a
     #         DummyOption. We need a default dummy because future steps require
     #         the option_spec field to be populated, even if just with a dummy.
-    nids = _get_initial_nids(segments, verbose=CFG.do_option_learning)
+    pnads = _get_initial_pnads(segments, verbose=CFG.do_option_learning)
 
-    # STEP 4: Learn side predicates for the operators and update NIDs. These
+    # STEP 4: Learn side predicates for the operators and update PNADs. These
     #         are predicates whose truth value becomes unknown (for *any*
     #         grounding not explicitly in effects) upon operator application.
     if CFG.learn_side_predicates:
-        _learn_nid_side_predicates(nids)
+        _learn_pnad_side_predicates(pnads)
 
-    # STEP 5: Learn options (option_learning.py) and update NIDs.
-    _learn_nid_options(nids)
+    # STEP 5: Learn options (option_learning.py) and update PNADs.
+    _learn_pnad_options(pnads)
```

Comment:
so PNADs are created before option learning, but PNADs have (not optional) option specs in them. before option learning, I guess these will just be default options, but is that weird? should be just make it optional? not sure

Reply:
Yup. I tried to make that clear in "STEP 3" near the top of nsrt_learning.py. I tried a bit to make it an optional field, but that made things weird with the actual clustering loop, since anyway we need a dummy option + specs for the purposes of running `unify_effects_and_options`. Don't think it's worth it, imo.